vumi/transports/httprpc/httprpc.py

```diff
@@ -193,6 +195,16 @@ def setup_transport(self):
             ],
             self.web_port)
 
+        self.status_detect = StatusEdgeDetector()
+
+    def add_status(self, **kw):
+        '''Publishes a status if it is not a repeat of the previously
+        published status.'''
+        status = TransportStatus(**kw)
+        if self.status_detect.check_status(status):
```

Comment:
Yes I didn't realise that `publish_status` worked that way, I thought that it accepted a `TransportStatus`. I'll change the edge detector to work that way.